src/Microsoft.NetCore.Analyzers/Core/Runtime/TestForEmptyStringsUsingStringLength.cs

```diff
@@ -66,6 +80,33 @@ private static void AnalyzeInvocationExpression(OperationAnalysisContext context
             }
         }
 
+        /// <summary>
+        /// Check to see if we have a equals or not equals expression where an empty string is being
+        /// compared.
+        /// </summary>
+        private static void AnalyzeBinaryExpression(OperationAnalysisContext context)
+        {
+            var binaryOperation = (IBinaryOperation)context.Operation;
```

Comment:
>(IBinaryOperation)context.Operation; [](start = 34, length = 36)

We access this method using the 'default' case in a switch statement. I assume that currently default == binary. I would add a protection against situations (maybe not possible today) where default != binary.

Reply:
Sure, I can add an assert.